test/staged.jl

```diff
@@ -135,3 +135,19 @@ f10502() = ()
 end
 @test f11982(Float32) == "Float32"
 @test f11982(Int32) == "Int32"
+
+# @generated functions that throw (shouldn't segfault or throw)
+module TestGeneratedThrow
+    using Base.Test
+
+    @generated function bar(x)
+        error("I'm not happy with type $x")
+    end
+
+    foo() = (bar(rand() > 0.5 ? 1 : 1.0); error("foo"))
+    function __init__()
```

Comment:
I'm not sure. I was unable to sufficiently confuse inference to cause a segfault without.